Competition2017/src/competition/subsystems/autonomous/DriveToHopperThenBoilerCommandGroup.java

```diff
@@ -0,0 +1,81 @@
+    public DoubleProperty distanceFromWallToBaseline;
+    public DoubleProperty distanceFromTurningPointToHopper;
+    public DoubleProperty distanceFromTurningPointToBoiler;
+    public DoubleProperty headingToFaceHopper;
+    public DoubleProperty waitTimeForFuelCollection;
+    public DoubleProperty headingToFaceBoiler;
+    public BooleanProperty goToBoiler;
+    
+    private class TimeoutCommand extends BaseCommand {
+        @Override
+        public void initialize() {
+            this.setTimeout(waitTimeForFuelCollection.get());
+        }
+
+        @Override
+        public void execute() {}
+        
+        public boolean isFinished(){
+            return this.isTimedOut();
+        }
+    }
+    
+    public DriveToHopperThenBoilerCommandGroup(
+            XPropertyManager propManager,
+            Provider<DriveForDistanceCommand> driveForDistanceProvider, 
+            Provider<DriveToOrientationCommand> driveToOrientationProvider){
+        DriveForDistanceCommand driveForDistance = driveForDistanceProvider.get();
+        DriveToOrientationCommand driveToOrientation = driveToOrientationProvider.get();
+
+        distanceFromWallToBaseline = propManager.createPersistentProperty("distanceFromWallToBaselineInInches", 37.25);
+        headingToFaceHopper = propManager.createPersistentProperty("headingToFaceHopperInDegrees", 90);
+        distanceFromTurningPointToHopper = propManager.createPersistentProperty("distanceFromTurningPointOnBaselineToHopperInInches", 43.25);
+        waitTimeForFuelCollection = propManager.createPersistentProperty("waitTimeForFuelCollectionInSeconds", 5);
+        goToBoiler = propManager.createPersistentProperty("goToBoiler", true);
+        headingToFaceBoiler = propManager.createPersistentProperty("headingToFaceBoilerInDegees", 70.2);
+        distanceFromTurningPointToHopper = propManager.createPersistentProperty("distanceFromTurningPointToHopperInInches", 72);
+
+        // move to turning point on baseline
+        driveForDistance.setDeltaDistance(distanceFromWallToBaseline.get());
+        this.addSequential(driveForDistance);
+
+        // change direction to face hopper
+        driveToOrientation.setHeading(headingToFaceHopper.get());
+        this.addSequential(driveToOrientation);
+
+        // move from the turning point on baseline to hopper
+        driveForDistance.setDeltaDistance(distanceFromTurningPointToHopper.get());
+        this.addSequential(driveForDistance);
+
+        // wait at the hopper to collect fuel
+        this.addSequential(new TimeoutCommand());
+
+        if(goToBoiler.get()){
+            // move back to the turning point on the baseline from the hopper
+            driveForDistance.setDeltaDistance(distanceFromTurningPointToHopper.get() * -1);
+            this.addSequential(driveForDistance);
+
+            // turn to face the boiler
+            driveToOrientation.setHeading(headingToFaceBoiler.get());
```

Comment:
same thing with this DriveToOrientation, you need to create multiple instances, one per unique thing you want to do.